feat(web): Implement keep this delete others for asset stacks

bdavis2-PCTY opened the pull request:
Adds a new button when viewing a stacked asset to "Keep this, delete others."

This will result in deleting all the assets that are in the stack, except for the one that is currently being viewed. 

![image](https://github.com/user-attachments/assets/79c964d9-32c0-442a-b292-9a59fc20e666)

bdavis2-PCTY committed:
Implement keep this delete others for asset stacks

bdavis2-PCTY committed:
Remove unused import

bdavis2-PCTY committed:
Code cleanup

danieldietzler commented on web/src/lib/components/asset-viewer/actions/keep-this-delete-others.svelte:
```diff
@@ -0,0 +1,21 @@
+  };
+</script>
+
+<MenuOption icon={mdiPinOutline} onClick={handleKeepThisDeleteOthers} text={'Keep this, delete others'} />
```
This text should be internationalized

bdavis2-PCTY replied:
Oops, thought I did this before making the PR.

Fix has been pushed.

danieldietzler commented on web/src/lib/utils/asset-utils.ts:
```diff
@@ -438,6 +439,32 @@ export const deleteStack = async (stackIds: string[]) => {
+
+  try {
+    const stack = await getStack({ id: stackId });
+    const assetToKeep = stack.assets.find(x => x.id === keepId);
+    const assetsToDeleteIds = stack.assets.filter(x => x.id !== keepId).map(x => x.id);
```
Please don't use random, single character variable names

bdavis2-PCTY replied:
Code has been updated with more meaningful variable names here.

bdavis2-PCTY committed:
Internationalize text

bdavis2-PCTY committed:
Change variable names

mertalev commented on web/src/lib/components/asset-viewer/actions/keep-this-delete-others.svelte:
```diff
@@ -0,0 +1,22 @@
+  export let onAction: OnAction;
+
+  const handleKeepThisDeleteOthers = async () => {
+    const keptAsset = await keepThisDeleteOthers(asset.id, stack.id);
```
You already have the asset and stack here, no? Just change the `keepThisDeleteOthers` to accept those instead of getting the IDs and querying for them again.

bdavis2-PCTY replied:
Function signature has been updated to accept the Asset & Stack objects instead of ids. Originally implemented this way as passing ids to the functions seemed to be the most common way elsewhere, but I completely agree that passing in the objects themselves is much more efficient.

mertalev commented on web/src/lib/utils/asset-utils.ts:
```diff
@@ -438,6 +439,32 @@ export const deleteStack = async (stackIds: string[]) => {
+  const $t = get(t);
+
+  try {
+    const stack = await getStack({ id: stackId });
+    const assetToKeep = stack.assets.find((asset) => asset.id === keepId);
```
You can remove these after changing the function signature.

bdavis2-PCTY replied:
Commit has been pushed to address this.

bdavis2-PCTY committed:
Change keepThisDeleteOthers function to accept asset and stack objects instead of ids

mertalev commented on web/src/lib/utils/asset-utils.ts:
```diff
@@ -438,6 +440,26 @@ export const deleteStack = async (stackIds: string[]) => {
+  try {
+    const assetsToDeleteIds = stack.assets.filter((asset) => asset.id !== keepAsset.id).map((asset) => asset.id);
+    await deleteStacks({ bulkIdsDto: { ids: [stack.id] } });
+    await deleteAssets({ assetBulkDeleteDto: { ids: assetsToDeleteIds } });
```
`deleteAssets` will permanently delete these assets if the trash feature is disabled, so you'll want to display a deletion warning and get user confirmation. There should be other examples of this in the code.

bdavis2-PCTY replied:
Good to know.  I'll look into doing this!

bdavis2-PCTY replied:
Confirmation dialog has been implemented when clicking "Keep this, delete others."
![image](https://github.com/user-attachments/assets/a917f880-2077-4118-ad8f-04e51125482a)

bdavis2-PCTY committed:
Add confirmation dialog for keep this, delete others

bdavis2-PCTY committed:
Merge branch 'main' into feature/keep-this-delete-others

jrasm91 committed:
feat: track when assets are added to an album (#13725)

danieldietzler committed:
refactor(server): telemetry env variables (#13705)

refactor(server)!: telemetry env variables

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

jrasm91 committed:
refactor(server): database config (#13730)

jrasm91 committed:
refactor(server): user create logic (#13728)

zackpollard committed:
fix: duplicated library scan jobs and api server library watch (#13734)

renovate[bot] committed:
chore(deps): update base-image to v20241025 (major) (#13736)

chore(deps): update base-image to v20241025

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

darrelhong committed:
fix: X years ago ordering (#13742)

* fix: order by fileCreatedAt to account for tz

* run make sql

dotlambda committed:
chore(ml): remove setuptools from dependencies (#13762)

bo0tzz committed:
feat: github actions workflow to fix formatting (#13768)

Sweebah committed:
fix: Add DB_DATA_LOCATION config. (#13754)

Add DB_DATA_LOCATION config.

Posted [on reddit](https://www.reddit.com/r/unRAID/comments/1f8wq45/immich_install_via_docker_compose_database_cannot/).

github-actions[bot] committed:
chore: version v1.119.0

alextran1502 committed:
chore(mobile): post release task (#13779)

alextran1502 committed:
feat(web): add job action to search page result (#13784)

renovate[bot] committed:
chore(deps): update dependency @types/node to ^20.17.0 (#13787)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update stumpylog/image-cleaner-action action to v0.9.0 (#13788)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

yozik04 committed:
docs: note about Nginx reverse proxy compatibility with Let's Encrypt (#13764)

* Nginx reverse proxy compatibility with Let's Encrypt

* Update reverse-proxy.md

* Update reverse-proxy.md

Co-authored-by: bo0tzz <git@bo0tzz.me>

* Priority of = is higher

* Update reverse-proxy.md

---------

Co-authored-by: bo0tzz <git@bo0tzz.me>

aviv926 committed:
docs: Add info about album sync&folder view (#12371)

* Album Sync

* npm run format:fix

* fix path

* info about folder view

* Sync albums

* npm run format:fix

* Add information about limitations of a property that appears in multiple albums

* npm run format:fix

* - to ->

* Update docs/docs/features/mobile-app.mdx

Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

* Update docs/docs/features/mobile-app.mdx

Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>
Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

github-actions[bot] committed:
chore: version v1.119.1

mertalev commented on i18n/en.json:
```diff
@@ -462,6 +462,7 @@
   "confirm": "Confirm",
   "confirm_admin_password": "Confirm Admin Password",
   "confirm_delete_shared_link": "Are you sure you want to delete this shared link?",
+  "confirm_keep_this_delete_others": "Are you sure you want to keep this asset and delete the others?",
```
```suggestion
  "confirm_keep_this_delete_others": "All other assets in the stack will be deleted except for this asset. Are you sure you want to continue?",
```

jrasm91 committed:
fix(server): handle N/A duration response from ffprobe (#13803)

mertalev commented on web/src/lib/utils/asset-utils.ts:
```diff
@@ -438,6 +440,26 @@ export const deleteStack = async (stackIds: string[]) => {
+
+  try {
+    const assetsToDeleteIds = stack.assets.filter((asset) => asset.id !== keepAsset.id).map((asset) => asset.id);
+    await deleteStacks({ bulkIdsDto: { ids: [stack.id] } });
+    await deleteAssets({ assetBulkDeleteDto: { ids: assetsToDeleteIds } });
```
It's better to switch the order of these two requests. If the delete fails, it will avoid the stack becoming unstacked.

bdavis2-PCTY replied:
Sorry for the delay. Good suggestion. Updated in next commit.

weblate committed:
chore(web): update translations (#13688)

Translate-URL: https://hosted.weblate.org/projects/immich/immich/ca/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/de/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/en_devel/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/es/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/et/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/fi/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/hr/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/id/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/it/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ko/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/lv/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ms/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ro/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ru/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sk/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sr_Cyrl/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sr_Latn/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sv/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/th/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/uk/
Translation: Immich/immich

Co-authored-by: Amirul Asri <amirulasrix@gmail.com>
Co-authored-by: Dean Cvjetanović <forteee@gmail.com>
Co-authored-by: Diego Saint Esteben <diego@saintesteben.me>
Co-authored-by: Florian Ostertag <florian.kuepper@gmail.com>
Co-authored-by: Hexie <hexterjexter@gmail.com>
Co-authored-by: Indrek Haav <IndrekHaav@users.noreply.hosted.weblate.org>
Co-authored-by: Jiri Grönroos <jiri.gronroos@iki.fi>
Co-authored-by: Junghyuk Kwon <kwon@junghy.uk>
Co-authored-by: Lab Forensic <recoverylabkor@gmail.com>
Co-authored-by: Leo Bottaro <github@leobottaro.com>
Co-authored-by: LeonardoCasarotto <leonardo.casar8@gmail.com>
Co-authored-by: Linerly <linerly@proton.me>
Co-authored-by: Max <mail@heavygale.de>
Co-authored-by: Michal Micech <michal.micech@gmail.com>
Co-authored-by: Miki Mrvos <medolino2009@gmail.com>
Co-authored-by: Mārtiņš Bruņenieks <martinsb@gmail.com>
Co-authored-by: Napat Srichan <napatsrichan2001@gmail.com>
Co-authored-by: Roger Veciana Rovira <rveciana@gmail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: qtm <qtm@users.noreply.hosted.weblate.org>
Co-authored-by: tux2000 <tux2k@web.de>
Co-authored-by: Вячеслав Лукьяненко <madeinchuguev@gmail.com>

zackpollard committed:
feat: mount checks on a folder level (#13801)

zackpollard committed:
fix: shutdown api process when another worker exits unexpectedly (#13802)

zackpollard committed:
refactor: better postgres connection param typing (#13800)

alextran1502 committed:
fix(mobile): Revert thumbnail image ui improvements  (#13655) (#13806)

Revert "chore(mobile): thumbnail image ui improvements  (#13655)"

This reverts commit 16f2364e932c8ae5ef539f949225dbb961834161.

jrasm91 committed:
refactor(server): worker cleanup (#13805)

zackpollard committed:
feat(web): library settings crontab link includes existing expression (#13807)

jrasm91 committed:
fix(server): keep system config transformations (#13796)

alextran1502 committed:
chore(docs): add feature to readme (#13794)

alextran1502 committed:
fix(mobile): thumbnail not filled area on tablet (#13808)

ITestInProd committed:
feat(web): disable opening image and library sub-items by default (#13729)

Disable opening image and library sub-items by default

Disable both the Image Settings and the External Library sub-items by default. This aligns with auth other settings sub-items showing as collapsed by default.

jrasm91 committed:
chore(server): telemetry env clean up (#13813)

jrasm91 committed:
refactor(server): cls (#13814)

jrasm91 committed:
refactor(server): env validation (#13817)

renovate[bot] committed:
chore(deps): update grafana/grafana docker tag to v11.3.0 (#13763)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update terraform cloudflare to v4.45.0 (#13820)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update base-image to v20241029 (major) (#13821)

chore(deps): update base-image to v20241029

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alextran1502 committed:
feat(web): scrubber label and animation (#13815)

* feat(web): scrubber label and animation

* tune x fly in distance

* refactor

* lint and minor fix

* fly height

MickLesk committed:
chore(docs): update german readme to latest version (#13824)

* chore(docs): update german readme to latest version

* formatting

renovate[bot] committed:
chore(deps): update node to v22 (major) (#13792)

* chore(deps): update node to v22

* fix: deprecated constructor

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Dietzler <mail@ddietzler.dev>

renovate[bot] committed:
fix(deps): update machine-learning (#13786)

jrasm91 committed:
refactor(server): remove unused on method (#13830)

renovate[bot] committed:
chore(deps): update typescript-projects (#13789)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alextran1502 committed:
fix(mobile): search page (#13833)

* fix(mobile): search page minor problems

* fix: flashing between search

* restore search size

* remove print statement

* linting

dvbthien committed:
fix(mobile): Reapply thumbnail image improvements and fix issue #13804 (#13835)

fix issue #13804

Co-authored-by: dvbthien <dvbthien@gmail.com>

zackpollard committed:
feat: built-in automatic database backups (#13773)

renovate[bot] committed:
chore(deps): update dependency @types/node to ^22.8.1 (#13854)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency node to v22 (#13855)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alextran1502 committed:
fix(mobile): not throwing error when cannot parse orientation value (#13853)

* fix(mobile): not throw when error in parsing the orientation

* fix(mobile): not throw when error in parsing the orientation

* linting

zackpollard committed:
chore: move library settings cron expression to setting select element (#13856)

alangrainger committed:
chore(docs): Add Immich Public Proxy to the Community Projects list (#13836)

Update community-projects.tsx

Add Immich Public Proxy to Community Projects

jrasm91 committed:
refactor(server): job discovery (#13838)

refactor(server): job discorvery

renovate[bot] committed:
chore(deps): update flutter (#13455)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alextran1502 committed:
chore(mobile): update flutter_web_auth (#13863)

mertalev committed:
feat(server): use tonemapx for software tone-mapping (#13785)

yashrajjain726 committed:
fix(mobile): do not removed not backup asset when selecting the correspond options (#13256)

* fixed the local ids selecting issue

* code: updated impl inside deleteLocalOnlyAssets

* fix: used png instead of jpg to maintain picture quality

* Revert "fix: used png instead of jpg to maintain picture quality"

This reverts commit 04f2ed54e40b454a057f97872836d71af50b06de.

* fix: update logic from code-review perspective

* refractor (mobile) : Dart fix applied

* fix (mobile) : Updated multi grid as per requirement

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

C-Otto committed:
fix(server): wrong image dimensions for RAW files (RAF, CR2) (also fixes face preview) (#13377)

jrasm91 committed:
refactor(server): immich worker (#13871)

refactor(server): immich work

danieldietzler committed:
chore: svelte 5 🎉  (#13738)

chore: svelte 5

Co-authored-by: Michel Heusschen <59014050+michelheusschen@users.noreply.github.com>
Co-authored-by: Ben McCann <322311+benmccann@users.noreply.github.com>

cfitzw committed:
feat: render asset path (#13873)

render asset path

danieldietzler committed:
fix: healthcheck if custom host is set (#13887)

renovate[bot] committed:
fix(deps): update dependency exiftool-vendored to v28.7.0 (#13790)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Pranay-Pandey committed:
fix: remove duplicateIds on unique assets (#13752)

renovate[bot] committed:
fix(deps): update machine-learning (#13919)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update node (#13918)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @types/node to ^22.8.5 (#13923)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jrasm91 committed:
refactor(server): get worker (#13927)

zackpollard committed:
chore: better observability for geodata import (#13931)

zackpollard committed:
refactor: config init event for first config load (#13930)

alextran1502 committed:
chore(mobile): upgrade gradle (#13901)

* chore(mobile): upgrade gradle

* update photo_manager

* migrate from kapt to ksp

pumps kotlin and other dependency version to fix build issues

* increase java heap size

---------

Co-authored-by: shenlong-tanwen <139912620+shalong-tanwen@users.noreply.github.com>

alextran1502 committed:
chore(web): scrubber animation (#13936)

* chore(web): scrubber animation

* minor styling

renovate[bot] committed:
chore(deps): update dependency @types/node to ^22.8.6 (#13937)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update typescript-projects (#13922)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Dietzler <mail@ddietzler.dev>

renovate[bot] committed:
chore(deps): update base-image to v20241105 (major) (#13826)

mmomjian committed:
docs: fail2ban community project (#13943)

fail2ban

alextran1502 committed:
chore(mobile): translation update (#13942)

weblate committed:
chore(web): update translations (#13810)

Co-authored-by: Benjamin Freeman <b.t.freeman16@gmail.com>
Co-authored-by: Bezruchenko Simon <worcposj44@gmail.com>
Co-authored-by: Carl Dressler <cstmth@gmail.com>
Co-authored-by: Diego Saint Esteben <diego@saintesteben.me>
Co-authored-by: Dmitry Banny <dj.icecore@gmail.com>
Co-authored-by: Dragonslayer <chybzik@gmail.com>
Co-authored-by: Fjuro <fjuro@alius.cz>
Co-authored-by: Florian Ostertag <florian.kuepper@gmail.com>
Co-authored-by: Hurricane-32 <rodrigorimo@hotmail.com>
Co-authored-by: Indrek Haav <IndrekHaav@users.noreply.hosted.weblate.org>
Co-authored-by: Iwwww <mike.yolgin@gmail.com>
Co-authored-by: JBP <weblate@1peer1boom.nl>
Co-authored-by: Jonas Grubbauer <nicjontrickshots@gmail.com>
Co-authored-by: Junghyuk Kwon <kwon@junghy.uk>
Co-authored-by: Kamil Warczek <kamilwarczek@gmail.com>
Co-authored-by: Kovács Ádám <kovaszftw@gmail.com>
Co-authored-by: Linerly <linerly@proton.me>
Co-authored-by: Mehmet Numan Çam <mncam051@gmail.com>
Co-authored-by: Miki Mrvos <medolino2009@gmail.com>
Co-authored-by: Polly Julien <julien.polly@laposte.net>
Co-authored-by: Rastislav Simcisko <21sd95@gmail.com>
Co-authored-by: Samet Güzeldemirci <s.g52@gmx.de>
Co-authored-by: Shawn <xiaxinx@gmail.com>
Co-authored-by: Stan P <g97d6liib@mozmail.com>
Co-authored-by: T.Nikolaou <tnikolaou@users.noreply.hosted.weblate.org>
Co-authored-by: TheGothian <github@dreamstate.nu>
Co-authored-by: fmis13 <fmis13@disroot.org>
Co-authored-by: pyccl <changcongliang@163.com>
Co-authored-by: qtm <qtm@users.noreply.hosted.weblate.org>
Co-authored-by: vytautas <immichtranslation.a03gn@simplelogin.com>
Co-authored-by: waclaw66 <waclaw66@seznam.cz>
Co-authored-by: xxzp3 <oliver@bastholm.dk>
Co-authored-by: Вячеслав Лукьяненко <madeinchuguev@gmail.com>
Co-authored-by: 李奕寯 <eugenelego88@gmail.com>

github-actions[bot] committed:
chore: version v1.120.0

danieldietzler committed:
fix: docker link (#13956)

slamp committed:
docs: improve custom-locations wording to be easier to read (#13849)

* Improve wording to make it easier to read custom-locations.md

It's only grammatical change

* Update docs/docs/guides/custom-locations.md

Co-authored-by: bo0tzz <git@bo0tzz.me>

* Update custom-locations.md

Revert to 'because of' and remove 'hard drive'

---------

Co-authored-by: bo0tzz <git@bo0tzz.me>

thariq-shanavas committed:
docs: Added a note about avoiding redundant database backups (#13958)

* Add note about built-in backups

* npm run format:fix

alextran1502 committed:
fix(server): cannot render email template (#13957)

bo0tzz committed:
chore: tidy up backup-and-restore.md (#13961)

danieldietzler committed:
docs: 50k stars (#13964)

alextran1502 committed:
chore(mobile): post release task (#13954)

yodatak committed:
docs: add backups to startup folders list (#13967)

Add the check of backups that is done on immich microservice of backups folder presence

dotlambda committed:
fix(server): allow starting backup through API and fix pg_dumpall args when using database URLs (#13970)

* fix(server): allow starting backup through API

* fix(server): fix pg_dumpall args when using database URLs

The database has to be specified using `-d`, unlike for pg_dump.

Scrumplex committed:
fix(server): database backups compatible with deduplication (#13965)

gzip --rsyncable has a slightly worse compression ratio, but allows for
efficient deduplication and, as the name implies, faster rsync
operations.

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

jrasm91 committed:
docs: update roadmap (#13984)

alextran1502 committed:
fix(mobile): video player not playing in full size on Android (#13986)

jrasm91 committed:
fix(web): time zone dependent test (#13859)

github-actions[bot] committed:
chore: version v1.120.1

alextran1502 committed:
chore(mobile): update isar build (#13987)

jrasm91 committed:
refactor: remove smart info table (#13985)

alextran1502 committed:
chore(mobile): post release tasks (#13989)

jrasm91 committed:
refactor(server): cron repository (#13988)

jrasm91 committed:
refactor(server): cron validation (#13990)

zackpollard committed:
chore: backups custom location and config file docs (#13996)

jrasm91 committed:
fix(server): support non-default postgres port when taking a backup (#13992)

zhaoterryy committed:
fix(server): thumbnail rotation when using embedded previews (#13948)

bo0tzz committed:
docs: make IGNORE_MOUNT_CHECK warning stronger (#14011)

michelheusschen committed:
fix(web): use locale for scrubber label when scrolling (#14012)

ben-basten committed:
chore(docs): roadmap SEO (#14024)

alextran1502 committed:
fix(mobile): make sure date locale is inititialized for some languages (#14035)

mcarbonne committed:
feat(web): stable json settings export (#14036)

* recursively sort json output (settings)

* fix format/lint/...g

Snowknight26 committed:
chore(web): Update the new version announcement text (#14001)

* Update en.json

* Update en.json

* Update en.json

Mraedis committed:
docs: Fix DCM docs link (#14059)

Fix DCM docs link

danieldietzler committed:
chore: add weblate requests (#14051)

gamescom15 committed:
docs: clarify file size impact in hardware-transcoding.md (#14049)

zackpollard committed:
fix(server): attempt to delete failed backups immediately after failure (#13995)

zackpollard committed:
docs: backup folder name is backups (#14073)

zackpollard committed:
fix: config updates not applying for job and storage template service (#14074)

dotlambda committed:
chore(ml): replace fastapi-slim with fastapi (#14091)

The two have been identical since version 0.112.0:
https://github.com/fastapi/fastapi/discussions/11525#discussioncomment-10219861

renovate[bot] committed:
chore(deps): update node (#14090)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update base-image to v20241112 (major) (#14088)

chore(deps): update base-image to v20241112

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zackpollard committed:
feat(server): use pg_dumpall version that matches the database version (#14083)

alextran1502 committed:
fix(server): correct rotation for common files (#14092)

* fix(server): correct rotation for common files

* fix: test:

* pr feedback

zackpollard committed:
fix(server): backup version checks not handling database versions correctly (#14102)

github-actions[bot] committed:
chore: version v1.120.2

alextran1502 committed:
chore(mobile): post release tasks (#14105)

renovate[bot] committed:
chore(deps): update redis:6.2-alpine docker digest to 77c6e37 (#14098)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update docker.io/redis:6.2-alpine docker digest to 77c6e37 (#14097)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump ytanikin/PRConventionalCommits from 1.2.0 to 1.3.0 (#13051)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update prom/prometheus docker digest to 2659f4c (#13928)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mertalev committed:
refactor(mobile): video controls (#14086)

* refactor video controls

* inline

* make mute icon const

* move placeholder to private widget

* adjust text width, move volume button slightly right

amorphobia committed:
chore(docs): encode db dump in UTF-8 without BOM for Windows (#13775)

renovate[bot] committed:
chore(deps): update docker.io/redis:6.2-alpine docker digest to eaba718 (#14113)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update redis:6.2-alpine docker digest to eaba718 (#14114)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alextran1502 committed:
fix(mobile): duration ui overflow (#14120)

* fix(mobile): duration ui overflow

* pr feedback

p2kmgcl committed:
feat: use dateTimeOriginal to calculate album date (#14119)

NicholasFlamy committed:
docs: Update TrueNAS docs for TrueNAS SCALE 24.10 (#14067)

* initial-docs-update

* add-info-about-external-libraries

johnstef99 committed:
fix(mobile): fix logout timeout (#14104)

* fix(mobile): add timeout to logout

* chore(mobile): refactor timeout durations

* feat(mobile): add loading state to logout button

* chore(mobile): format authentication.provider.dart

* chore: format

* chore: revert settings.json change

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

mcarbonne committed:
feat: add minimal devcontainer setup (#14038)

* add minimal devcontainer setup

* fix Makefile & update doc

* fix Makefile

* add warning regarding devcontainer + add newline at EOF

mertalev committed:
feat(mobile): new video slider ui (#14126)

lukasdotcom committed:
fix(server): Some MTS videos fail to generate thumbnail (#14134)

* Stop skipping of all frames in MTS video

* Only skip flag for mts videos

* Fix lint checks

* Adds test

* Add comment for why flag is removed

alextran1502 committed:
chore(web): migration svelte 5 syntax (#13883)

duckimann committed:
fix(web): textarea autogrow height (#13983)

fix(web): remove album assetGridWidth & globalWidth

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

Pranay-Pandey committed:
fix: Routing back button in sharedLinks page (#13703)

* fix: go back to last page from shared links page. Handle albums page from shared links page routing

* add default route for sharing

* chore: remove redundant import

* remove unnecessary comment

---------

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

michelheusschen committed:
fix(web): ensure current asset index stays within bounds (#14013)

michelheusschen committed:
fix(web): saving pasted coordinates (#14143)

alextran1502 committed:
chore(web): move enum out of .svelte file (#14144)

* chore(web): clean up todo task

* chore(web): move enums out of .svelte file

michelheusschen committed:
fix(web): update description height when navigating between assets (#14145)

michelheusschen committed:
fix(web): allow selecting people after clearing search options (#14146)

alextran1502 committed:
fix(mobile): Android local notification failes to invoke (#14155)

* fix(mobile): local notification failed to invoke

* add proguard rules

IMBeniamin committed:
feat(web): Added tag button to the context menu in the favorites page (#14156)

Added tag action to the context menu in the favorites page when selection is active

renovate[bot] committed:
chore(deps): update prom/prometheus docker digest to 3b9b2a1 (#14154)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michelheusschen committed:
fix(web): prevent infinite loop when modifying stacked asset (#14162)

Tiefseetauchner committed:
fix(cli): Concurrency not fully using queue potential (#11828)

mertalev committed:
refactor(mobile): backup info box (#14171)

split up backup info box into separate widgets

michelheusschen committed:
refactor(web): remove reactivity triggers (#14164)

michelheusschen committed:
fix(web): don't refresh panorama viewer when modifying asset (#14163)

weathondev committed:
feat: adding photo & video storage space to server stats (#14125)

* expose detailed user storage stats + display them in the storage per user table

* chore: openapi & sql

* fix: fix test stubs

* fix: formatting errors, e2e test and server test

* fix: upper lower case typo in spec file

---------

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

mmomjian committed:
docs: get asset owner by ID from database (#14174)

asset owner

mertalev committed:
fix(mobile): unnecessary rebuilds from partner share notifier (#14170)

* fix unnecessary notifications

* move equality function

* sort by id

* use same comparison for initial and later queries

michelheusschen committed:
fix: show tags when viewing stacked assets (#14199)

fix: refresh tags when viewing stacked assets

renovate[bot] committed:
chore(deps): pin mcr.microsoft.com/devcontainers/typescript-node docker tag to dc2c365 (#14124)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mertalev committed:
fix(mobile): use sets in album refresh, concurrent futures (#14193)

* use sets in album sync, concurrent futures

* batch excluded asset IDs

* update test

* take advantage of sets in Recents check

* move log statement

* smaller diff

johnstef99 committed:
fix(mobile): Dismissible menus (#14192)

* chore(mobile): make all search filters dismissible

* chore(mobile): make ImmichAppBarDialog dismissible

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

bdavis2-PCTY committed:
Update i18n/en.json

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

bdavis2-PCTY committed:
Reverse delete stacks/assets order

bdavis2-PCTY committed:
Backmerge (#1)

* feat: track when assets are added to an album (#13725)

* refactor(server): telemetry env variables (#13705)

refactor(server)!: telemetry env variables

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

* refactor(server): database config (#13730)

* refactor(server): user create logic (#13728)

* fix: duplicated library scan jobs and api server library watch (#13734)

* chore(deps): update base-image to v20241025 (major) (#13736)

chore(deps): update base-image to v20241025

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix: X years ago ordering (#13742)

* fix: order by fileCreatedAt to account for tz

* run make sql

* chore(ml): remove setuptools from dependencies (#13762)

* feat: github actions workflow to fix formatting (#13768)

* fix: Add DB_DATA_LOCATION config. (#13754)

Add DB_DATA_LOCATION config.

Posted [on reddit](https://www.reddit.com/r/unRAID/comments/1f8wq45/immich_install_via_docker_compose_database_cannot/).

* chore: version v1.119.0

* chore(mobile): post release task (#13779)

* feat(web): add job action to search page result (#13784)

* chore(deps): update dependency @types/node to ^20.17.0 (#13787)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update stumpylog/image-cleaner-action action to v0.9.0 (#13788)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* docs: note about Nginx reverse proxy compatibility with Let's Encrypt (#13764)

* Nginx reverse proxy compatibility with Let's Encrypt

* Update reverse-proxy.md

* Update reverse-proxy.md

Co-authored-by: bo0tzz <git@bo0tzz.me>

* Priority of = is higher

* Update reverse-proxy.md

---------

Co-authored-by: bo0tzz <git@bo0tzz.me>

* docs: Add info about album sync&folder view (#12371)

* Album Sync

* npm run format:fix

* fix path

* info about folder view

* Sync albums

* npm run format:fix

* Add information about limitations of a property that appears in multiple albums

* npm run format:fix

* - to ->

* Update docs/docs/features/mobile-app.mdx

Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

* Update docs/docs/features/mobile-app.mdx

Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>
Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

* chore: version v1.119.1

* fix(server): handle N/A duration response from ffprobe (#13803)

* chore(web): update translations (#13688)

Translate-URL: https://hosted.weblate.org/projects/immich/immich/ca/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/de/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/en_devel/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/es/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/et/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/fi/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/hr/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/id/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/it/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ko/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/lv/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ms/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ro/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/ru/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sk/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sr_Cyrl/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sr_Latn/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/sv/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/th/
Translate-URL: https://hosted.weblate.org/projects/immich/immich/uk/
Translation: Immich/immich

Co-authored-by: Amirul Asri <amirulasrix@gmail.com>
Co-authored-by: Dean Cvjetanović <forteee@gmail.com>
Co-authored-by: Diego Saint Esteben <diego@saintesteben.me>
Co-authored-by: Florian Ostertag <florian.kuepper@gmail.com>
Co-authored-by: Hexie <hexterjexter@gmail.com>
Co-authored-by: Indrek Haav <IndrekHaav@users.noreply.hosted.weblate.org>
Co-authored-by: Jiri Grönroos <jiri.gronroos@iki.fi>
Co-authored-by: Junghyuk Kwon <kwon@junghy.uk>
Co-authored-by: Lab Forensic <recoverylabkor@gmail.com>
Co-authored-by: Leo Bottaro <github@leobottaro.com>
Co-authored-by: LeonardoCasarotto <leonardo.casar8@gmail.com>
Co-authored-by: Linerly <linerly@proton.me>
Co-authored-by: Max <mail@heavygale.de>
Co-authored-by: Michal Micech <michal.micech@gmail.com>
Co-authored-by: Miki Mrvos <medolino2009@gmail.com>
Co-authored-by: Mārtiņš Bruņenieks <martinsb@gmail.com>
Co-authored-by: Napat Srichan <napatsrichan2001@gmail.com>
Co-authored-by: Roger Veciana Rovira <rveciana@gmail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: qtm <qtm@users.noreply.hosted.weblate.org>
Co-authored-by: tux2000 <tux2k@web.de>
Co-authored-by: Вячеслав Лукьяненко <madeinchuguev@gmail.com>

* feat: mount checks on a folder level (#13801)

* fix: shutdown api process when another worker exits unexpectedly (#13802)

* refactor: better postgres connection param typing (#13800)

* fix(mobile): Revert thumbnail image ui improvements  (#13655) (#13806)

Revert "chore(mobile): thumbnail image ui improvements  (#13655)"

This reverts commit 16f2364e932c8ae5ef539f949225dbb961834161.

* refactor(server): worker cleanup (#13805)

* feat(web): library settings crontab link includes existing expression (#13807)

* fix(server): keep system config transformations (#13796)

* chore(docs): add feature to readme (#13794)

* fix(mobile): thumbnail not filled area on tablet (#13808)

* feat(web): disable opening image and library sub-items by default (#13729)

Disable opening image and library sub-items by default

Disable both the Image Settings and the External Library sub-items by default. This aligns with auth other settings sub-items showing as collapsed by default.

* chore(server): telemetry env clean up (#13813)

* refactor(server): cls (#13814)

* refactor(server): env validation (#13817)

* chore(deps): update grafana/grafana docker tag to v11.3.0 (#13763)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update terraform cloudflare to v4.45.0 (#13820)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update base-image to v20241029 (major) (#13821)

chore(deps): update base-image to v20241029

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* feat(web): scrubber label and animation (#13815)

* feat(web): scrubber label and animation

* tune x fly in distance

* refactor

* lint and minor fix

* fly height

* chore(docs): update german readme to latest version (#13824)

* chore(docs): update german readme to latest version

* formatting

* chore(deps): update node to v22 (major) (#13792)

* chore(deps): update node to v22

* fix: deprecated constructor

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Dietzler <mail@ddietzler.dev>

* fix(deps): update machine-learning (#13786)

* refactor(server): remove unused on method (#13830)

* chore(deps): update typescript-projects (#13789)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix(mobile): search page (#13833)

* fix(mobile): search page minor problems

* fix: flashing between search

* restore search size

* remove print statement

* linting

* fix(mobile): Reapply thumbnail image improvements and fix issue #13804 (#13835)

fix issue #13804

Co-authored-by: dvbthien <dvbthien@gmail.com>

* feat: built-in automatic database backups (#13773)

* chore(deps): update dependency @types/node to ^22.8.1 (#13854)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update dependency node to v22 (#13855)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix(mobile): not throwing error when cannot parse orientation value (#13853)

* fix(mobile): not throw when error in parsing the orientation

* fix(mobile): not throw when error in parsing the orientation

* linting

* chore: move library settings cron expression to setting select element (#13856)

* chore(docs): Add Immich Public Proxy to the Community Projects list (#13836)

Update community-projects.tsx

Add Immich Public Proxy to Community Projects

* refactor(server): job discovery (#13838)

refactor(server): job discorvery

* chore(deps): update flutter (#13455)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(mobile): update flutter_web_auth (#13863)

* feat(server): use tonemapx for software tone-mapping (#13785)

* fix(mobile): do not removed not backup asset when selecting the correspond options (#13256)

* fixed the local ids selecting issue

* code: updated impl inside deleteLocalOnlyAssets

* fix: used png instead of jpg to maintain picture quality

* Revert "fix: used png instead of jpg to maintain picture quality"

This reverts commit 04f2ed54e40b454a057f97872836d71af50b06de.

* fix: update logic from code-review perspective

* refractor (mobile) : Dart fix applied

* fix (mobile) : Updated multi grid as per requirement

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

* fix(server): wrong image dimensions for RAW files (RAF, CR2) (also fixes face preview) (#13377)

* refactor(server): immich worker (#13871)

refactor(server): immich work

* chore: svelte 5 🎉  (#13738)

chore: svelte 5

Co-authored-by: Michel Heusschen <59014050+michelheusschen@users.noreply.github.com>
Co-authored-by: Ben McCann <322311+benmccann@users.noreply.github.com>

* feat: render asset path (#13873)

render asset path

* fix: healthcheck if custom host is set (#13887)

* fix(deps): update dependency exiftool-vendored to v28.7.0 (#13790)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix: remove duplicateIds on unique assets (#13752)

* fix(deps): update machine-learning (#13919)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update node (#13918)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update dependency @types/node to ^22.8.5 (#13923)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* refactor(server): get worker (#13927)

* chore: better observability for geodata import (#13931)

* refactor: config init event for first config load (#13930)

* chore(mobile): upgrade gradle (#13901)

* chore(mobile): upgrade gradle

* update photo_manager

* migrate from kapt to ksp

pumps kotlin and other dependency version to fix build issues

* increase java heap size

---------

Co-authored-by: shenlong-tanwen <139912620+shalong-tanwen@users.noreply.github.com>

* chore(web): scrubber animation (#13936)

* chore(web): scrubber animation

* minor styling

* chore(deps): update dependency @types/node to ^22.8.6 (#13937)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update typescript-projects (#13922)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Dietzler <mail@ddietzler.dev>

* chore(deps): update base-image to v20241105 (major) (#13826)

* docs: fail2ban community project (#13943)

fail2ban

* chore(mobile): translation update (#13942)

* chore(web): update translations (#13810)

Co-authored-by: Benjamin Freeman <b.t.freeman16@gmail.com>
Co-authored-by: Bezruchenko Simon <worcposj44@gmail.com>
Co-authored-by: Carl Dressler <cstmth@gmail.com>
Co-authored-by: Diego Saint Esteben <diego@saintesteben.me>
Co-authored-by: Dmitry Banny <dj.icecore@gmail.com>
Co-authored-by: Dragonslayer <chybzik@gmail.com>
Co-authored-by: Fjuro <fjuro@alius.cz>
Co-authored-by: Florian Ostertag <florian.kuepper@gmail.com>
Co-authored-by: Hurricane-32 <rodrigorimo@hotmail.com>
Co-authored-by: Indrek Haav <IndrekHaav@users.noreply.hosted.weblate.org>
Co-authored-by: Iwwww <mike.yolgin@gmail.com>
Co-authored-by: JBP <weblate@1peer1boom.nl>
Co-authored-by: Jonas Grubbauer <nicjontrickshots@gmail.com>
Co-authored-by: Junghyuk Kwon <kwon@junghy.uk>
Co-authored-by: Kamil Warczek <kamilwarczek@gmail.com>
Co-authored-by: Kovács Ádám <kovaszftw@gmail.com>
Co-authored-by: Linerly <linerly@proton.me>
Co-authored-by: Mehmet Numan Çam <mncam051@gmail.com>
Co-authored-by: Miki Mrvos <medolino2009@gmail.com>
Co-authored-by: Polly Julien <julien.polly@laposte.net>
Co-authored-by: Rastislav Simcisko <21sd95@gmail.com>
Co-authored-by: Samet Güzeldemirci <s.g52@gmx.de>
Co-authored-by: Shawn <xiaxinx@gmail.com>
Co-authored-by: Stan P <g97d6liib@mozmail.com>
Co-authored-by: T.Nikolaou <tnikolaou@users.noreply.hosted.weblate.org>
Co-authored-by: TheGothian <github@dreamstate.nu>
Co-authored-by: fmis13 <fmis13@disroot.org>
Co-authored-by: pyccl <changcongliang@163.com>
Co-authored-by: qtm <qtm@users.noreply.hosted.weblate.org>
Co-authored-by: vytautas <immichtranslation.a03gn@simplelogin.com>
Co-authored-by: waclaw66 <waclaw66@seznam.cz>
Co-authored-by: xxzp3 <oliver@bastholm.dk>
Co-authored-by: Вячеслав Лукьяненко <madeinchuguev@gmail.com>
Co-authored-by: 李奕寯 <eugenelego88@gmail.com>

* chore: version v1.120.0

* fix: docker link (#13956)

* docs: improve custom-locations wording to be easier to read (#13849)

* Improve wording to make it easier to read custom-locations.md

It's only grammatical change

* Update docs/docs/guides/custom-locations.md

Co-authored-by: bo0tzz <git@bo0tzz.me>

* Update custom-locations.md

Revert to 'because of' and remove 'hard drive'

---------

Co-authored-by: bo0tzz <git@bo0tzz.me>

* docs: Added a note about avoiding redundant database backups (#13958)

* Add note about built-in backups

* npm run format:fix

* fix(server): cannot render email template (#13957)

* chore: tidy up backup-and-restore.md (#13961)

* docs: 50k stars (#13964)

* chore(mobile): post release task (#13954)

* docs: add backups to startup folders list (#13967)

Add the check of backups that is done on immich microservice of backups folder presence

* fix(server): allow starting backup through API and fix pg_dumpall args when using database URLs (#13970)

* fix(server): allow starting backup through API

* fix(server): fix pg_dumpall args when using database URLs

The database has to be specified using `-d`, unlike for pg_dump.

* fix(server): database backups compatible with deduplication (#13965)

gzip --rsyncable has a slightly worse compression ratio, but allows for
efficient deduplication and, as the name implies, faster rsync
operations.

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

* docs: update roadmap (#13984)

* fix(mobile): video player not playing in full size on Android (#13986)

* fix(web): time zone dependent test (#13859)

* chore: version v1.120.1

* chore(mobile): update isar build (#13987)

* refactor: remove smart info table (#13985)

* chore(mobile): post release tasks (#13989)

* refactor(server): cron repository (#13988)

* refactor(server): cron validation (#13990)

* chore: backups custom location and config file docs (#13996)

* fix(server): support non-default postgres port when taking a backup (#13992)

* fix(server): thumbnail rotation when using embedded previews (#13948)

* docs: make IGNORE_MOUNT_CHECK warning stronger (#14011)

* fix(web): use locale for scrubber label when scrolling (#14012)

* chore(docs): roadmap SEO (#14024)

* fix(mobile): make sure date locale is inititialized for some languages (#14035)

* feat(web): stable json settings export (#14036)

* recursively sort json output (settings)

* fix format/lint/...g

* chore(web): Update the new version announcement text (#14001)

* Update en.json

* Update en.json

* Update en.json

* docs: Fix DCM docs link (#14059)

Fix DCM docs link

* chore: add weblate requests (#14051)

* docs: clarify file size impact in hardware-transcoding.md (#14049)

* fix(server): attempt to delete failed backups immediately after failure (#13995)

* docs: backup folder name is backups (#14073)

* fix: config updates not applying for job and storage template service (#14074)

* chore(ml): replace fastapi-slim with fastapi (#14091)

The two have been identical since version 0.112.0:
https://github.com/fastapi/fastapi/discussions/11525#discussioncomment-10219861

* chore(deps): update node (#14090)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update base-image to v20241112 (major) (#14088)

chore(deps): update base-image to v20241112

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* feat(server): use pg_dumpall version that matches the database version (#14083)

* fix(server): correct rotation for common files (#14092)

* fix(server): correct rotation for common files

* fix: test:

* pr feedback

* fix(server): backup version checks not handling database versions correctly (#14102)

* chore: version v1.120.2

* chore(mobile): post release tasks (#14105)

* chore(deps): update redis:6.2-alpine docker digest to 77c6e37 (#14098)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update docker.io/redis:6.2-alpine docker digest to 77c6e37 (#14097)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): bump ytanikin/PRConventionalCommits from 1.2.0 to 1.3.0 (#13051)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): update prom/prometheus docker digest to 2659f4c (#13928)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* refactor(mobile): video controls (#14086)

* refactor video controls

* inline

* make mute icon const

* move placeholder to private widget

* adjust text width, move volume button slightly right

* chore(docs): encode db dump in UTF-8 without BOM for Windows (#13775)

* chore(deps): update docker.io/redis:6.2-alpine docker digest to eaba718 (#14113)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* chore(deps): update redis:6.2-alpine docker digest to eaba718 (#14114)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix(mobile): duration ui overflow (#14120)

* fix(mobile): duration ui overflow

* pr feedback

* feat: use dateTimeOriginal to calculate album date (#14119)

* docs: Update TrueNAS docs for TrueNAS SCALE 24.10 (#14067)

* initial-docs-update

* add-info-about-external-libraries

* fix(mobile): fix logout timeout (#14104)

* fix(mobile): add timeout to logout

* chore(mobile): refactor timeout durations

* feat(mobile): add loading state to logout button

* chore(mobile): format authentication.provider.dart

* chore: format

* chore: revert settings.json change

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

* feat: add minimal devcontainer setup (#14038)

* add minimal devcontainer setup

* fix Makefile & update doc

* fix Makefile

* add warning regarding devcontainer + add newline at EOF

* feat(mobile): new video slider ui (#14126)

* fix(server): Some MTS videos fail to generate thumbnail (#14134)

* Stop skipping of all frames in MTS video

* Only skip flag for mts videos

* Fix lint checks

* Adds test

* Add comment for why flag is removed

* chore(web): migration svelte 5 syntax (#13883)

* fix(web): textarea autogrow height (#13983)

fix(web): remove album assetGridWidth & globalWidth

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

* fix: Routing back button in sharedLinks page (#13703)

* fix: go back to last page from shared links page. Handle albums page from shared links page routing

* add default route for sharing

* chore: remove redundant import

* remove unnecessary comment

---------

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

* fix(web): ensure current asset index stays within bounds (#14013)

* fix(web): saving pasted coordinates (#14143)

* chore(web): move enum out of .svelte file (#14144)

* chore(web): clean up todo task

* chore(web): move enums out of .svelte file

* fix(web): update description height when navigating between assets (#14145)

* fix(web): allow selecting people after clearing search options (#14146)

* fix(mobile): Android local notification failes to invoke (#14155)

* fix(mobile): local notification failed to invoke

* add proguard rules

* feat(web): Added tag button to the context menu in the favorites page (#14156)

Added tag action to the context menu in the favorites page when selection is active

* chore(deps): update prom/prometheus docker digest to 3b9b2a1 (#14154)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix(web): prevent infinite loop when modifying stacked asset (#14162)

* fix(cli): Concurrency not fully using queue potential (#11828)

* refactor(mobile): backup info box (#14171)

split up backup info box into separate widgets

* refactor(web): remove reactivity triggers (#14164)

* fix(web): don't refresh panorama viewer when modifying asset (#14163)

* feat: adding photo & video storage space to server stats (#14125)

* expose detailed user storage stats + display them in the storage per user table

* chore: openapi & sql

* fix: fix test stubs

* fix: formatting errors, e2e test and server test

* fix: upper lower case typo in spec file

---------

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

* docs: get asset owner by ID from database (#14174)

asset owner

* fix(mobile): unnecessary rebuilds from partner share notifier (#14170)

* fix unnecessary notifications

* move equality function

* sort by id

* use same comparison for initial and later queries

* fix: show tags when viewing stacked assets (#14199)

fix: refresh tags when viewing stacked assets

* chore(deps): pin mcr.microsoft.com/devcontainers/typescript-node docker tag to dc2c365 (#14124)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

* fix(mobile): use sets in album refresh, concurrent futures (#14193)

* use sets in album sync, concurrent futures

* batch excluded asset IDs

* update test

* take advantage of sets in Recents check

* move log statement

* smaller diff

* fix(mobile): Dismissible menus (#14192)

* chore(mobile): make all search filters dismissible

* chore(mobile): make ImmichAppBarDialog dismissible

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

---------

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>
Co-authored-by: Jason Rasmussen <jason@rasm.me>
Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>
Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>
Co-authored-by: Zack Pollard <zackpollard@ymail.com>
Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Darrel <39296145+darrelhong@users.noreply.github.com>
Co-authored-by: Robert Schütz <mail@dotlambda.de>
Co-authored-by: bo0tzz <git@bo0tzz.me>
Co-authored-by: Sweebah <mistri.boy@gmail.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Alex <alex.tran1502@gmail.com>
Co-authored-by: Jevgeni Kiski <yozik04@gmail.com>
Co-authored-by: aviv926 <51673860+aviv926@users.noreply.github.com>
Co-authored-by: Weblate (bot) <hosted@weblate.org>
Co-authored-by: Amirul Asri <amirulasrix@gmail.com>
Co-authored-by: Dean Cvjetanović <forteee@gmail.com>
Co-authored-by: Diego Saint Esteben <diego@saintesteben.me>
Co-authored-by: Florian Ostertag <florian.kuepper@gmail.com>
Co-authored-by: Hexie <hexterjexter@gmail.com>
Co-authored-by: Indrek Haav <IndrekHaav@users.noreply.hosted.weblate.org>
Co-authored-by: Jiri Grönroos <jiri.gronroos@iki.fi>
Co-authored-by: Junghyuk Kwon <kwon@junghy.uk>
Co-authored-by: Lab Forensic <recoverylabkor@gmail.com>
Co-authored-by: Leo Bottaro <github@leobottaro.com>
Co-authored-by: LeonardoCasarotto <leonardo.casar8@gmail.com>
Co-authored-by: Linerly <linerly@proton.me>
Co-authored-by: Max <mail@heavygale.de>
Co-authored-by: Michal Micech <michal.micech@gmail.com>
Co-authored-by: Miki Mrvos <medolino2009@gmail.com>
Co-authored-by: Mārtiņš Bruņenieks <martinsb@gmail.com>
Co-authored-by: Napat Srichan <napatsrichan2001@gmail.com>
Co-authored-by: Roger Veciana Rovira <rveciana@gmail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: qtm <qtm@users.noreply.hosted.weblate.org>
Co-authored-by: tux2000 <tux2k@web.de>
Co-authored-by: Вячеслав Лукьяненко <madeinchuguev@gmail.com>
Co-authored-by: Justin <5438593+ITestInProd@users.noreply.github.com>
Co-authored-by: CanbiZ <47820557+MickLesk@users.noreply.github.com>
Co-authored-by: Daniel Dietzler <mail@ddietzler.dev>
Co-authored-by: dvbthien <89862334+dvbthien@users.noreply.github.com>
Co-authored-by: dvbthien <dvbthien@gmail.com>
Co-authored-by: Alan Grainger <alan.grainger@gmail.com>
Co-authored-by: Yashraj Jain <yashrajjain726@gmail.com>
Co-authored-by: Carsten Otto <git@c-otto.de>
Co-authored-by: Michel Heusschen <59014050+michelheusschen@users.noreply.github.com>
Co-authored-by: Ben McCann <322311+benmccann@users.noreply.github.com>
Co-authored-by: cfitzw <20694057+cfitzw@users.noreply.github.com>
Co-authored-by: Pranay Pandey <79053599+Pranay-Pandey@users.noreply.github.com>
Co-authored-by: shenlong-tanwen <139912620+shalong-tanwen@users.noreply.github.com>
Co-authored-by: Matthew Momjian <50788000+mmomjian@users.noreply.github.com>
Co-authored-by: Benjamin Freeman <b.t.freeman16@gmail.com>
Co-authored-by: Bezruchenko Simon <worcposj44@gmail.com>
Co-authored-by: Carl Dressler <cstmth@gmail.com>
Co-authored-by: Dmitry Banny <dj.icecore@gmail.com>
Co-authored-by: Dragonslayer <chybzik@gmail.com>
Co-authored-by: Fjuro <fjuro@alius.cz>
Co-authored-by: Hurricane-32 <rodrigorimo@hotmail.com>
Co-authored-by: Iwwww <mike.yolgin@gmail.com>
Co-authored-by: JBP <weblate@1peer1boom.nl>
Co-authored-by: Jonas Grubbauer <nicjontrickshots@gmail.com>
Co-authored-by: Kamil Warczek <kamilwarczek@gmail.com>
Co-authored-by: Kovács Ádám <kovaszftw@gmail.com>
Co-authored-by: Mehmet Numan Çam <mncam051@gmail.com>
Co-authored-by: Polly Julien <julien.polly@laposte.net>
Co-authored-by: Rastislav Simcisko <21sd95@gmail.com>
Co-authored-by: Samet Güzeldemirci <s.g52@gmx.de>
Co-authored-by: Shawn <xiaxinx@gmail.com>
Co-authored-by: Stan P <g97d6liib@mozmail.com>
Co-authored-by: T.Nikolaou <tnikolaou@users.noreply.hosted.weblate.org>
Co-authored-by: TheGothian <github@dreamstate.nu>
Co-authored-by: fmis13 <fmis13@disroot.org>
Co-authored-by: pyccl <changcongliang@163.com>
Co-authored-by: vytautas <immichtranslation.a03gn@simplelogin.com>
Co-authored-by: waclaw66 <waclaw66@seznam.cz>
Co-authored-by: xxzp3 <oliver@bastholm.dk>
Co-authored-by: 李奕寯 <eugenelego88@gmail.com>
Co-authored-by: slamp <slaamp@gmail.com>
Co-authored-by: Thariq Shanavas <thariqshanavas@gmail.com>
Co-authored-by: yodatak <mryodatak@gmail.com>
Co-authored-by: Sefa Eyeoglu <contact@scrumplex.net>
Co-authored-by: Terry Zhao <zhao.terryy@gmail.com>
Co-authored-by: Ben <45583362+ben-basten@users.noreply.github.com>
Co-authored-by: mcarbonne <46689813+mcarbonne@users.noreply.github.com>
Co-authored-by: Snowknight26 <Snowknight26@users.noreply.github.com>
Co-authored-by: Joren Guillaume <guillaume.joren@gmail.com>
Co-authored-by: gamescom15 <10470023+gamescom15@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Xuesong <amorphobia@users.noreply.github.com>
Co-authored-by: Pablo Molina <gitju@pablomolina.me>
Co-authored-by: Nicholas Flamy <30300649+NicholasFlamy@users.noreply.github.com>
Co-authored-by: John Stef <john@johnstef.com>
Co-authored-by: Lukas <lukas@lschaefer.xyz>
Co-authored-by: Đức <facoder2k2@gmail.com>
Co-authored-by: Ben <35833890+IMBeniamin@users.noreply.github.com>
Co-authored-by: Lena Tauchner <48085877+Tiefseetauchner@users.noreply.github.com>
Co-authored-by: weathondev <weathon.dev@gmail.com>

bdavis2-PCTY commented:
As you all can see, I totally messed up the git commit log by accident. 

I've made a new branch to clear all that here: https://github.com/immich-app/immich/pull/14217/files

bo0tzz commented:
For future reference, you don't need to create new PRs to clean up a mess, just fix it locally and then force-push to your PR branch.
